### ISSUE TITLE
Test for connection during dry-run before steps are run

### ISSUE THREAD

**Issue by mhadam** [snowplow/sql-runner#137]
It seems that the connection is tested on each query, which would be excessive.

Add the connection test for each target in `run.go`:
https://github.com/snowplow/sql-runner/blob/develop/sql_runner/run.go#L255

**Comment by alexanderdean** [comment]
Is this a bug in the v0.7.0 series @mhadam ?

**Comment by mhadam** [comment]
Yes. Not a show-stopper, it's does what it needs to, it just gets called more than it needs to be.

**Comment by alexanderdean** [comment]
Cool, scheduled into the next minor release.
